ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,227 @@ namespace ngraph
+                    int32_t input_rank = input_shape.rank().get_length();
+                    uint64_t input_h = input_shape[input_rank - 2].get_length();
+                    uint64_t input_w = input_shape[input_rank - 1].get_length();
+                    std::cout << input_rank << "," << input_h << "," << input_w << std::endl;
+
+                    auto auto_pad = ngraph::op::PadType::EXPLICIT;
+                    ngraph::Shape pad_begin, pad_end;
+                    get_paddings(node, pad_begin, pad_end, auto_pad);
+
+                    if (global_pooling ||
+                        (adaptive && std::any_of(kernel_shape.begin(),
+                                                 kernel_shape.end(),
+                                                 [](int32_t i) { return i == 1; })))
+                    {
+                        if (pooling_type == "max")
+                        {
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::MaxPool>(
+                                    data,
+                                    ngraph::Strides({1, 1}),
+                                    ngraph::Shape{0, 0}, // FIXME pads_begin
+                                    ngraph::Shape{0, 0}, // pads_end
+                                    ngraph::Shape{input_h, input_w})},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            auto axes = ngraph::opset6::Constant::create(
+                                ngraph::element::i64, {2}, {input_rank - 2, input_rank - 1});
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
+                                {"Out"});
+                        }
                     }
-                    else if (pooling_type == "avg" &&
-                             (global_pooling ||
-                              (adaptive && all_of(kernel_shape.begin(),
-                                                  kernel_shape.end(),
-                                                  [](int32_t s) { return s == 1; }))))
+                    else if (adaptive)
                     {
-                        // TODO : resolve axes according to rank
-                        auto axes =
-                            ngraph::opset6::Constant::create(ngraph::element::i64, {2}, {2, 3});
-                        return node.default_single_output_mapping(
-                            {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
-                            {"Out"});
+                        uint64_t pool_size_Height, pool_size_Width;
+                        if (kernel_shape.size() == 1)
+                        {
+                            pool_size_Height = pool_size_Width = kernel_shape[0];
```

Comment:
not covered by unit test. If can't generate model, suggest to add note that it is not tested on real models

Reply:
@ceciliapeng2011 please add coverage test